lib/jobs/generate-tag.js

```diff
@@ -66,36 +71,53 @@ function generateTagJobFactory(
                 })
                 .uniq()
                 .value();
-            return [ Promise.map(catalogTypes, function(type) {
+
+            if (self.nodes) {
+                return Promise.map(self.nodes, function(nodeId) {
+                    return Promise.map(catalogTypes, function(type) {
+                        return waterline.catalogs.findMostRecent({
+                            node: nodeId,
+                            source: type
+                        });
+                    }).then(processCatalogs.bind(null, tags, nodeId));
+                });
+            }
+
+            return Promise.map(catalogTypes, function(type) {
```

Comment:
If you agree to assign the singular nodeId into self.nodes, then we can completely remove this code block.